### PR TITLE
[Storage] Fix up `QueueProperties` docstring

### PR DESCRIPTION
**Pull request by vincenttran-msft**
As title states

### CODE REVIEW

File: sdk/storage/azure-storage-queue/azure/storage/queue/_models.py [hunk 1]
```diff
@@ -455,22 +455,25 @@ def _extract_data_cb(self, messages: Any) -> Tuple[str, List[QueueMessage]]:
...
-    name: str
+    name: Optional[str]
     """The name of the queue."""
     metadata: Optional[Dict[str, str]]
     """A dict containing name-value pairs associated with the queue as metadata."""
     approximate_message_count: Optional[int]
     """The approximate number of messages contained in the queue."""
 
     def __init__(self, **kwargs: Any) -> None:
+        self.name = kwargs.get('name', None)
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_models.py:476]: Not necessary:
```suggestion
        self.name = kwargs.get('name')
```

File: sdk/storage/azure-storage-queue/azure/storage/queue/_models.py [hunk 2]
```diff
@@ -455,22 +455,25 @@ def _extract_data_cb(self, messages: Any) -> Tuple[str, List[QueueMessage]]:
 class QueueProperties(DictMixin):
     """Queue Properties.
 
-    :keyword str name:
+    :keyword Optional[str] name:
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_models.py:458]: Can we also update `_from_generated`?

Comment (vincenttran-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_models.py:458]: Hi Anna, maybe this is obvious (and it is going right over my head), but what is the change we are looking for here?

Here is the definition for `_from_generated`:
``` 
@classmethod
    def _from_generated(cls, generated: Any) -> Self:
        props = cls()
        props.name = generated.name
        props.metadata = generated.metadata
        return props
```

Do we want to omit setting `props.name`, or what type of update are we looking for here?

File: sdk/storage/azure-storage-queue/azure/storage/queue/_models.py [hunk 3]
```diff
@@ -455,22 +455,25 @@ def _extract_data_cb(self, messages: Any) -> Tuple[str, List[QueueMessage]]:
...
         The name of the queue.
     :keyword Optional[Dict[str, str]] metadata:
         A dict containing name-value pairs associated with the queue as metadata.
         This var is set to None unless the include=metadata param was included
         for the list queues operation.
+    :keyword Optional[int] approximate_message_count:
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_models.py:464]: This isn't actually a keyword. The keyword is actually the header name 'x-ms-approximate-messages-count', though I don't think we should document that....

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/azure-storage-queue/azure/storage/queue/_models.py [hunk 4]
```diff
@@ -471,6 +470,7 @@ class QueueProperties(DictMixin):
     """The approximate number of messages contained in the queue."""
 
     def __init__(self, **kwargs: Any) -> None:
+        self.name = None #type: ignore [assignment]
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_models.py:473]: As discussed offline, we will type-ignore this for now and look to modify and correct the behavior further as a separate item: https://github.com/Azure/azure-sdk-for-python/issues/33389

**Pull request merged**